utils/frame/generate-bags/src/lib.rs

```diff
@@ -0,0 +1,227 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Support code to ease the process of generating bag thresholds.
+//!
+//! NOTE: this assume the runtime implements [`pallet_staking::Config`], as it requires an
+//! implementation of the traits [`frame_support::traits::Currency`] and
+//! [`frame_support::traits::CurrencyToVote`].
+//!
+//! The process of adding bags to a runtime requires only four steps.
+//!
+//! 1. Update the runtime definition.
+//!
+//!    ```ignore
+//!    parameter_types!{
+//!         pub const BagThresholds: &'static [u64] = &[];
+//!    }
+//!
+//!    impl pallet_bags_list::Config for Runtime {
+//!         // <snip>
+//!         type BagThresholds = BagThresholds;
+//!    }
+//!    ```
+//!
+//! 2. Write a little program to generate the definitions. This program exists only to hook together
+//! the runtime definitions with the various calculations here. Take a look at
+//! _utils/frame/node-runtime_ for an example.
+//!
+//! 3. Run that program:
+//!
+//!    ```sh,notrust
+//!    $ cargo run -p node-runtime-generate-bags -- output.rs
+//!    ```
+//!
+//! 4. Update the runtime definition.
+//!
+//!    ```diff,notrust
+//!    + mod output;
+//!    - pub const BagThresholds: &'static [u64] = &[];
+//!    + pub const BagThresholds: &'static [u64] = &output::THRESHOLDS;
+//!    ```
+
+use frame_election_provider_support::VoteWeight;
+use frame_support::traits::Get;
+use std::{
+	io::Write,
+	path::{Path, PathBuf},
+};
+
+/// Compute the existential weight for the specified configuration.
+///
+/// Note that this value depends on the current issuance, a quantity known to change over time.
+/// This makes the project of computing a static value suitable for inclusion in a static,
+/// generated file _excitingly unstable_.
+fn existential_weight<T: pallet_staking::Config>() -> VoteWeight {
```

Comment:
personally I think it would be easier to just pass the total issuance and the minimal balance to the cli rather than querying from state which involve a difficult setup of the state.

(sidenote: maybe we can provide a macro which is `generate_bags!` which takes 3 argument: number of bags, minimum balance and total issuance as a raw integers and generates the bags, similar to the reward_curve macro, it can be used like this:
```rust
const BAGS: &[u64] = generate_bags!(
number_of_bags = 200,
minimum_balance = 1, 
total_issuance = 100_000_000
);
```
But we can do later if wanted.
)

Reply:
2ee478cd6f0250994acfaf2dc0ab947a8926e91c

Reply:
the root docs of generate-bags can be update too I think, but implementation is good to me

Reply:
35a94719e6ca16b747ef03cb01aa80f0e13fb679